### ISSUE TITLE
False positive with numerals in match_str_case_mismatch

### ISSUE THREAD

**Issue by Herschel** [rust-lang/rust-clippy#7863]
Lint name:
match_str_case_mismatch

I tried this code:

https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=1efc4425e26da357ac3f0734bd11df1a

```rust
match s.to_ascii_lowercase().as_str() {
        "8x8" => 8,
        "16x16" => 16,
        _ => 0,
    };
```

I expected to see this happen: No lints triggered.

Instead, this happened:
```
error: this `match` arm has a differing case than its expression
 --> src/main.rs:4:9
  |
4 |         "8x8" => 8,
  |         ^^^^^
  |
  = note: `#[deny(clippy::match_str_case_mismatch)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#match_str_case_mismatch
help: consider changing the case of this arm to respect `to_ascii_lowercase`
```

### Meta

**Rust version (`rustc -Vv`):**
```
rustc 1.58.0-nightly (514b38779 2021-10-22)
binary: rustc
commit-hash: 514b3877956dc594823106b66c164f8cdbc8b3da
commit-date: 2021-10-22
host: x86_64-pc-windows-msvc
release: 1.58.0-nightly
LLVM version: 13.0.0
```
